src/backlight/backlight.c

```diff
@@ -378,26 +397,33 @@ int main(int argc, char *argv[]) {
 
         if (streq(argv[1], "load")) {
                 _cleanup_free_ char *value = NULL;
-                const char *clamp;
+                bool clamp;
 
                 if (shall_restore_state() == 0)
                         return EXIT_SUCCESS;
 
                 if (!validate_device(udev, device))
                         return EXIT_SUCCESS;
 
-                r = read_one_line_file(saved, &value);
-                if (r < 0) {
+                clamp = shall_clamp(device);
 
-                        if (r == -ENOENT)
+                r = read_one_line_file(saved, &value);
+                if (r == -ENOENT) {
+                        if (!clamp)
                                 return EXIT_SUCCESS;
 
+                        /* not saved: fake "0" to trigger brightness clamping if necessary */
+                        value = strdup("0");
```

Comment:
hmm, why no use  `udev_device_get_sysattr_value(device, "brightness", …` here to query the currently set brightness? That sounds like the much better approach, than always resetting the brightness to lowest

Reply:
Makes sense.